src/api.js

```diff
@@ -138,6 +138,54 @@ var schema = function(validator, options) {
   };
 };
 
+
+/**
+ * Validate query-string against query.
+ *
+ * options:
+ * {
+ *   param1: /.../,               // Reg-exp pattern
+ *   param2(val) { return "..." } // Function, returns message if invalid
+ * }
+ *
+ * Query-string options not specified in options will not be allowed. But it's
+ * optional if a request carries any query-string parameters at all.
+ */
+var queryValidator = function(options = {}) {
+  return function(req, res, next) {
+    let errors = [];
+    _.forEach(req.query || {}, (value, key) => {
+      let pattern = options[key];
+      if (!pattern) {
+        // Allow the bewit key, it's used in signed strings
+        if (key === 'bewit') {
+          return;
+        }
+        // Unsupported option
+        errors.push('Query-string parameter: ' + key + ' is not supported!');
+      }
+      if (pattern instanceof RegExp) {
+        if (!pattern.test(value)) {
+          errors.push('Query-string parameter: ' + key + '="' + value +
+                      '" does not match expression: ' + pattern.toString());
+        }
+      } else {
+        let msg = pattern(value);
+        if (typeof(msg) === 'string') {
+          errors.push('Query-string parameter: ' + key + '="' + value +
+                      '" is not valid, error: ' + msg);
+        }
+      }
+    });
+    if (errors.length > 0) {
+      return res.status(400).json({
+        message: errors.join('\n'),
+      });
+    }
+    return next();
+  };
+};
+
```

Comment:
I like validation being built in from the start!  This seems like a good approach to me, I like it.